test/org/sagebionetworks/bridge/models/accounts/ConsentStatusTest.java

```diff
@@ -0,0 +1,129 @@
+    public void hashCodeEquals() {
+        EqualsVerifier.forClass(ConsentStatus.class).allFieldsShouldBeUsed().verify(); 
+    }
+    
+    // Will be stored as JSON in the the session, via the User object, so it must serialize.
+    @Test
+    public void canSerialize() throws Exception {
+        ConsentStatus status = new ConsentStatus("Name", "GUID", true, true, true);
+
+        String json = BridgeObjectMapper.get().writeValueAsString(status);
+        JsonNode node = BridgeObjectMapper.get().readTree(json);
+        
+        assertEquals("Name", node.get("name").asText());
+        assertEquals("GUID", node.get("subpopulationGuid").asText());
+        assertTrue(node.get("required").asBoolean());
+        assertTrue(node.get("consented").asBoolean());
+        assertTrue(node.get("mostRecentConsent").asBoolean());
+        assertEquals("ConsentStatus", node.get("type").asText());
+        
+        ConsentStatus status2 = BridgeObjectMapper.get().readValue(json, ConsentStatus.class);
+        assertEquals(status, status2);
+    }
+
+    @Test
+    public void forSubpopulation() {
+        Subpopulation subpop = Subpopulation.create();
+        subpop.setGuid("test");
+        
+        assertNull(ConsentStatus.forSubpopulation(statuses, subpop));
+        
+        ConsentStatus status1 = new ConsentStatus("Name1", "foo", false, false, false);
+        ConsentStatus status2 = new ConsentStatus("Name2", "test", false, false, false);
+        statuses.add(status1);
+        statuses.add(status2);
+        
+        assertEquals(status2, ConsentStatus.forSubpopulation(statuses, subpop));
+    }
+
+    @Test
+    public void isUserConsented() {
+        assertFalse(ConsentStatus.isUserConsented(statuses));
+        
+        statuses.add(REQUIRED_UNSIGNED);
+        statuses.add(REQUIRED_SIGNED_CURRENT);
+        statuses.add(OPTIONAL_SIGNED_CURRENT);
+        assertFalse(ConsentStatus.isUserConsented(statuses));
+        
+        statuses.clear();
+        statuses.add(REQUIRED_SIGNED_CURRENT);
+        statuses.add(REQUIRED_UNSIGNED);
+        statuses.add(OPTIONAL_UNSIGNED);
+        assertFalse(ConsentStatus.isUserConsented(statuses));
+
+        statuses.clear();
+        statuses.add(REQUIRED_SIGNED_CURRENT);
+        statuses.add(REQUIRED_SIGNED_OBSOLETE);
+        statuses.add(OPTIONAL_UNSIGNED);
+        assertTrue(ConsentStatus.isUserConsented(statuses));
+    }
+
+    @Test
+    public void isConsentCurrent() {
+        assertFalse(ConsentStatus.isConsentCurrent(statuses));
+        
+        statuses.add(REQUIRED_SIGNED_CURRENT);
+        statuses.add(REQUIRED_SIGNED_OBSOLETE);
+        statuses.add(OPTIONAL_UNSIGNED);
+        assertFalse(ConsentStatus.isConsentCurrent(statuses));
+        
+        statuses.clear();
+        statuses.add(REQUIRED_SIGNED_CURRENT);
+        statuses.add(REQUIRED_SIGNED_CURRENT);
+        statuses.add(OPTIONAL_UNSIGNED);
+        assertTrue(ConsentStatus.isConsentCurrent(statuses));
+    }
```

Comment:
Do we have the concept of optional and obselete? If so, do we want to add it to our test case here for completeness?